### ISSUE TITLE
Reported false positive errors on properties specified via assignment rather than the @property decorator

### ISSUE THREAD

**Issue by dmach** [python/mypy#15957]
**Bug Report**

Properties decorated with `@property` seem to be working fine.
Properties specified via assignment and instantiating a property class (or an inherited class from it) do not and mypy outputs errors.

**To Reproduce**

```python
class A:
    def getter(self) -> bool:
        return True

    attr: bool = property(getter)


class MyProperty(property):
    pass


class B:
    def getter(self) -> bool:
        return True

    attr: bool = MyProperty(getter)


from typing import *
print(get_type_hints(A.attr.fget))
# {'return': <class 'bool'>}
```

**Expected Behavior**

mypy should not report invalid type on assignment, because we're assigning a property that has a getter of a correct type
also, mypy shouldn't complain about `"bool" has no attribute "fget"` in A.attr.fget because we're working with a property, not the actual value from an instance.

**Actual Behavior**
```
error: Incompatible types in assignment (expression has type "property", variable has type "bool")  [assignment]
error: Incompatible types in assignment (expression has type "MyProperty", variable has type "bool")  [assignment]
error: "bool" has no attribute "fget"  [attr-defined]
```

**Your Environment**

- Mypy version used: 1.5.0
- Mypy command-line flags: <none>
- Mypy configuration options from `mypy.ini` (and other config files): <none / defaults to openSUSE Tumbleweed>
- Python version used: 3.11

**Comment by AlexWaygood** [comment]
Duplicate of #8083
